### PR TITLE
Recommend using bundler to ensure the correct dependencies are installed

### PR DESCRIPTION
**Pull request by matt-allan**
Hi Hoodies,

I was looking for a starter issue and saw the comments on [this post](https://github.com/hoodiehq/hood.ie/issues/230) about having trouble setting up jekyll locally.  Then I tried and had the same issues :)

I was able to get everything working correctly by doing a `bundle install` and then using `bundle exec jekyll *` instead of `jekyll *`.

Since the jekyll-redirect-from and jekyll-paginate plugins are already specified in the github-pages gem, `bundle install` will always install the correct versions.

I wasn't sure why `bundle exec` worked better, but here is what I found [on the bundler website:](http://bundler.io/)

> In some cases, running executables without bundle exec may work, if the executable happens to be installed in your system and does not pull in any gems that conflict with your bundle.
> 
> However, this is unreliable and is the source of considerable pain. Even if it looks like it works, it may not work in the future or on another machine.

### CODE REVIEW

**Comment by gr2m** [comment]
Thanks @yuloh, I can confirm that bundle exec also worked for me when I run into this. 
LGTM :+1:

File: README.md [hunk 1]
```diff
@@ -92,12 +91,12 @@ Please fork the website and send a Pull Request for your contribution. Here is <
 
 We will review the Pull Request (PR) and merge or comment it.
 
-**Content changes and typos**   
+**Content changes and typos**
```
Comment (gr2m) [inline README.md]: can you leave the white space? It’s needed for line breaks. Same in line 99

**Comment by matt-allan** [comment]
Sorry, editor preferences :smile:

**Comment by gr2m** [comment]
I’ve another question, it’s been a while since my ruby days :) We have `Gemfile.lock` in `.gitignore`. The [bundler website](http://bundler.io/) suggests to add it to git though: `git add Gemfile Gemfile.lock`. What do you think would be recommended for us?

**Comment by janl** [comment]
If we do this, we need to add how to install bundler to the instructions ;)

```
> bundle install
-bash: bundle: command not found
```

Also, we are using https://github.com/jekyll/jekyll-redirect-from and it’s not defined in our `Gemfile`, we should all fix this in one go.

**Comment by janl** [comment]
@yuloh great initiative! :)

**Comment by steveklabnik** [comment]
Yes, this is correct. `bundle exec` will ensure that only the gems in your `Gemfile` and their dependencies are in the path.

> I’ve another question, it’s been a while since my ruby days :) We have Gemfile.lock in .gitignore. The bundler website suggests to add it to git though: git add Gemfile Gemfile.lock. What do you think would be recommended for us?

It should be in ignore for libraries, but not for applications. This site is an application, so it should be in source control, not ignored.

**Comment by matt-allan** [comment]
Another option is to not commit the lock file and parse the versions.json provided by github, which would ensure the local version of the github-pages gem is always up to date.  It's explained [in the jekyll docs](https://jekyllrb.com/docs/github-pages/) under the "Use the github-pages gem" heading.  If we commit the `gemfile.lock` we will just need to run `bundle update` regularly.

> If we do this, we need to add how to install bundler to the instructions ;)

Good point.  I will push an update to the readme.

> Also, we are using https://github.com/jekyll/jekyll-redirect-from and it’s not defined in our Gemfile, we should all fix this in one go.

I thought the same thing initially but it's actually [installed by the github pages gem](https://github.com/github/pages-gem/blob/master/lib/github-pages/dependencies.rb#L26).  All of the github pages gem dependencies are listed [here](https://pages.github.com/versions/).  Since github pages builds with the --safe option, any dependencies not listed in the github-pages gem are ignored.

**Comment by janl** [comment]
> I thought the same thing initially but it's actually installed by the github pages gem. 

@yuloh perfect!

File: README.md [hunk 2]
```diff
@@ -3,11 +3,19 @@ Hood.ie: the Website for the Hoodie Open Source Project
...
 ## Getting started
 
+### Installation
+
 To get started check out the recent version and type `npm install`.
-The default task (just type `grunt`) will fire up a local server at `localhost:1337` with livereload and dev Sass compiling (including sourcemap and nested output).
 
-This doesn't serve the actual jekyll page, though. To do that, run `jekyll serve --watch --drafts` as well, which will run the full page at `localhost:4000`, using the styles from the grunt task.
-You may need to `gem install jekyll-redirect-from jekyll-paginate` beforehand.
+The Hoodie website is built with [Jekyll](https://jekyllrb.com/), Jekyll requires ruby version 2.0.0 or higher.  Open a terminal and type `ruby --version` to check if ruby is installed and up to date.  If not, follow the [official download instructions](https://www.ruby-lang.org/en/downloads/).  Now install [bundler](http://bundler.io/) by running `gem install bundler` on the command line.
```
Comment (matt-allan) [inline README.md:10]: These instructions are paraphrased from the [github pages instructions](https://help.github.com/articles/setting-up-your-pages-site-locally-with-jekyll/#preparing-to-install-the-github-pages-gem).

**Comment by matt-allan** [comment]
I got some clarification on the downsides of using the `versions.json` approach [over here](https://github.com/github/pages-gem/issues/261).  Every time you do `bundle exec jekyll serve` it will make an HTTP call, which isn't really ideal.

I'm going to update this pull request to version the Gemfile.lock, since that seems like the best way to go.  It does mean contributors will occasionally need to run `bundle update github-pages` to keep the local version in sync.

File: Gemfile [hunk 3]
```diff
@@ -1,2 +1,2 @@
 source 'https://rubygems.org'
-gem 'github-pages'
+gem 'github-pages', group: :jekyll_plugins
```
Comment (matt-allan) [inline Gemfile:2]: This change allows the github-pages gem to set jekyll config settings, like enabling Github flavored markdown.  The full config it's using is [in this file](https://github.com/github/pages-gem/blob/master/lib/github-pages/configuration.rb).

**Comment by gr2m** [comment]
Tested it locally and worked flawlessly, great work everyone :) LGTM

**Comment by gr2m** [comment]
@janl can you double check before we merge?

**Comment by gr2m** [comment]
merged via 7b040c2, 44575e0, 412af70, a5d6910

**Pull request closed without merging**